browser/src/control/Control.NavigatorPanel.ts

```diff
@@ -426,35 +426,46 @@ class NavigatorPanel extends SidebarBase {
 		data: any,
 		builder: JSBuilder,
 	): void {
-		let searchTerm = '';
-		// Update results tab
-		if (object.id === 'navigator-search-button') {
-			searchTerm = (
-				document.getElementById('navigator-search-input') as HTMLInputElement
-			).value;
-			super.callback('edit', 'activate', { id: 'Find' }, searchTerm, builder);
-		} else if (object.id === 'navigator-search') {
-			searchTerm = data;
-			super.callback(
-				objectType,
-				eventType,
-				{ id: 'Find' },
-				searchTerm,
-				builder,
-			);
+		if (!['navigator-search-button', 'navigator-search'].includes(object.id)) {
```

Comment:
Somehow the new code is more complex to analyze what it does. Can we put list in the comment?

// 1. If we click on search button -> just pass to callback ...
// 2. If we requested search -> switch to result tab and send message ...
// 3. If we got ... then we do ...

Or we could also restructure it in some smaller functions (so it names and explains the code at the same time) ?

Reply:
Separated into 2 functions.